iceoryx_utils/include/iceoryx_utils/internal/concurrent/smart_lock.hpp

```diff
@@ -140,11 +141,14 @@ class smart_lock
     ///         if ( iter != vectorGuard->end() )
     ///             vectorGuard->erase(iter);
     ///     }
-    Proxy GetScopeGuard() const;
+    Proxy GetScopeGuard() const noexcept;
+
+    /// @brief Returns a copy of the underlying object
+    T GetCopy() const noexcept;
 
   private:
     T base;
-    MutexType lock;
+    mutable MutexType lock;
```

Comment:
This is just `mutable` because `GetCopy()` is `const`? Why not remove the `const`?

Reply:
Since `GetCopy` does return just a copy and does not change the contents of the class. This is a typical approach for thread-safe constructs since the `const` is a contract for the user that the object state is not changed (which is the case here).
A mutex is excluded from the state since it is a helper struct to ensure the atomic access to the state of the object.